runtime/compiler/z/codegen/J9S390PrivateLinkage.cpp

```diff
@@ -3051,6 +3051,97 @@ void TR::J9S390JNILinkage::acquireVMAccessMask(TR::Node * callNode, TR::Register
    }
 
 
+#ifdef J9VM_INTERP_ATOMIC_FREE_JNI
+void
+TR::J9S390JNILinkage::releaseVMAccessMaskAtomicFree(TR::Node * callNode,
+                                                    TR::Register * methodMetaDataVirtualRegister,
+                                                    TR::Register * tempReg1)
+   {
+   TR_J9VMBase *fej9 = (TR_J9VMBase *)fe();
+   TR::CodeGenerator* cg = self()->cg();
+   TR::Compilation* comp = self()->comp();
+
+   // Store 1 into vmthread->inNative
+   generateRIInstruction(cg, TR::InstOpCode::getLoadHalfWordImmOpCode(), callNode, tempReg1, 1);
+   generateRXInstruction(cg, TR::InstOpCode::getStoreOpCode(), callNode, tempReg1,
+                         generateS390MemoryReference(methodMetaDataVirtualRegister, offsetof(J9VMThread, inNative), cg));
```

Comment:
Can this be done with an `MVI`?

Reply:
Changed to use `MVGHI` or `MVHI` depending on bitness (via the `getMoveHalfWordImmOpCode` query).